Frameworks/AudioToolbox/AudioConverter.mm

```diff
@@ -16,28 +17,160 @@
 
 #import <AudioToolbox/AudioConverter.h>
 #import <StubReturn.h>
+#import <NSBundleInternal.h>
+#import <NSLogging.h>
+#import "AssertARCEnabled.h"
+
+#include <COMIncludes.h>
+#import <wrl\client.h>
+#import <wrl\wrappers\corewrappers.h>
+#include <mfapi.h>
+#include <mfidl.h>
+#include <mferror.h>
+#include <wmcodecdsp.h>
+#include <COMIncludes_End.h>
+
+using namespace Microsoft::WRL;
+using namespace Microsoft::WRL::Wrappers;
+
+@interface AudioConverter : NSObject
+@property (atomic) ComPtr<IMFTransform> transform;
+@property (atomic) float sizeChangeMultiplier;
+- (instancetype)initWithSourceFormat:(const AudioStreamBasicDescription*)srcFormat
+                   destinationFormat:(const AudioStreamBasicDescription*)destFormat;
+- (ComPtr<IMFTransform>)getTransform;
+- (float)getSizeChangeMultiplier;
+@end
+
+static const wchar_t* TAG = L"AudioConverter";
+
+#define RETURN_AUDIOERR_IF_FAILED_WITH_MSG(hr, msg) \
+    if (FAILED(hr)) {                               \
+        NSTraceInfo(TAG, @"%@", msg);               \
+        return kAudioConverterErr_UnspecifiedError; \
+    }
+
+@implementation AudioConverter
+- (instancetype)initWithSourceFormat:(const AudioStreamBasicDescription*)srcFormat
+                   destinationFormat:(const AudioStreamBasicDescription*)destFormat {
+    if (self = [super init]) {
+        ComPtr<IUnknown> spTransformUnk;
+
+        RETURN_NULL_IF_FAILED(
```

Comment:
I believe ARC should handle it as we are returning NULL. Also went ahead and explicitly declared self = nil for the failure case